### PR TITLE
Pass billing details to auth flow URL

### PR DESCRIPTION
**Pull request by mats-stripe**
## Summary

In order to populate billing details on a payment method created in the web auth flow, we need to pass in all the billing details to the auth flow URL. We now pass the following query params to the auth flow URL: 

- `billingDetails[name]`
- `billingDetails[address][city]`
- `billingDetails[address][country]`
- `billingDetails[address][line1]`
- `billingDetails[address][line2]`
- `billingDetails[address][postal_code]`
- `billingDetails[address][state]`

## Motivation

Fix web auth flow billing details

## Testing

Unit tests and manual testing:

Screenshot 2024-10-30 at 2 10 46 PM

## Changelog

N/a

### CODE REVIEW

File: StripePaymentSheet/StripePaymentSheet/Source/PaymentSheet/ViewControllers/PaymentMethodFormViewController.swift [hunk 1]
```diff
@@ -225,7 +225,7 @@ extension PaymentMethodFormViewController {
             return PhoneNumber.fromE164(defaultPhoneNumber)?.number
         }()
         let prefillDetails = ElementsSessionContext.PrefillDetails(
-            email: instantDebitsFormElement?.email ?? configuration.defaultBillingDetails.name,
+            email: instantDebitsFormElement?.email ?? configuration.defaultBillingDetails.email,
```
Comment (mats-stripe) [inline StripePaymentSheet/StripePaymentSheet/Source/PaymentSheet/ViewControllers/PaymentMethodFormViewController.swift:228]: Drive-by bug fix 😅

File: StripeFinancialConnections/StripeFinancialConnections/Source/Web/FinancialConnectionsWebFlowViewController.swift [hunk 2]
```diff
@@ -424,30 +425,75 @@ private extension FinancialConnectionsWebFlowViewController {
...
+                    parameters.append("billingDetails[address][postal_code]=\(postalCode)")
+                }
+                if let state = address.state, !state.isEmpty {
+                    parameters.append("billingDetails[address][state]=\(state)")
+                }
+            }
+        }
+
+        // Join all values with an &, and URL encode.
+        guard let result = parameters.joined(separator: "&").addingPercentEncoding(withAllowedCharacters: .urlHostAllowed) else {
```
Comment (tillh-stripe) [inline StripeFinancialConnections/StripeFinancialConnections/Source/Web/FinancialConnectionsWebFlowViewController.swift:493]: Nit: Can you add a comment about why the URL encoding is necessary here?

Comment (mats-stripe) [reply StripeFinancialConnections/StripeFinancialConnections/Source/Web/FinancialConnectionsWebFlowViewController.swift:494]: Done in https://github.com/stripe/stripe-ios/pull/4203/commits/04ff788a61c1b4a2eda6e782d05b75def9452af5

**Review by tillh-stripe — APPROVED** [review]

**Review by tillh-stripe — DISMISSED** [review]

**Review by tillh-stripe — APPROVED** [review]
